src/components/SearchPhotos/SearchPhotos.js

```diff
@@ -0,0 +1,118 @@
+import React, { Component } from 'react';
+import { withRouter } from 'react-router-dom';
+import axios from 'axios';
+import Loader from 'react-loader-spinner';
+import InfiniteScroll from 'react-infinite-scroll-component';
+import { getSearchResults } from 'utils/api';
+import Modal from 'components/Modal';
+import {
+  GalleryImage,
+  GalleryItem,
+  LoadingSpinner,
+  Message,
+} from 'pages/Photos/PhotoList.styles';
+import { Gallery } from './SearchPhotos.styles';
+import { breakpointColumns } from 'utils/helper';
+
+export class SearchPhotos extends Component {
+  state = {
+    photos: [],
+    page: 1,
+    perPage: 25,
+    show: false,
+    index: null,
+    isLoading: false,
+    hasMore: true,
+    error: null,
+  };
+  showModal = (index) => {
+    this.setState({ show: true, index });
+  };
+
+  hideModal = () => {
+    this.setState({ show: false });
+  };
```

Comment:
you can make the modal method like this, always set the index to a number, if the number is -1 don't show the modal